src/main.js

```diff
@@ -1,43 +1,196 @@
+    }
+    
+    return data;
+}
+
+function formatParams(params, module, funcCallerParam) {
+    function formatError(err) {
+        return {
+            message: err.message,
+            stack: err.stack,
+        };
+    }
+
+    const funcCaller = funcCallerParam || {};
+    const result = [];
+    const metadata = {};
+
+    if (tracer && tracer.currentRootSpan && tracer.currentRootSpan.traceId) {
+        metadata.traceId = tracer.currentRootSpan.traceId;
+    }
+
+    if (typeof params[0] === 'string') {
+        result[0] = params[0];
+    } else if (params[0] instanceof Error) {
+        result[0] = params[0].message;
+        metadata.error = formatError(params[0]);
+    } else {
+        result[0] = util.inspect(params[0], {
+            compact: prettyLogs,
+            colors: prettyLogs,
+            depth: null,
+            breakLength: Infinity,
+        });
+    }
+
+    for (let i = 1; i < params.length; i++) {
+        if (params[i] instanceof Error) {
+            metadata.error = formatError(params[i]);
+        } else if (typeof params[i] === 'object') {
+            if (!metadata.extra) {
+                metadata.extra = {};
+            }
+
+            Object.assign(metadata.extra, params[i]);
+        }
+    }
+
+    if(module) {
+        metadata.module = module;
+    }
+
+    Object.assign(metadata, funcCaller);
+    result.push(metadata);
+    return result;
+}
+
+function debugAndFuncCaller(funcCaller, module, ...params) {
+    const formattedParams = formatParams(params, module, funcCaller);
+    winstonLogger.debug(...formattedParams);
+}
+
+function errorAndFuncCaller(funcCaller, module, ...params) {
+    const formattedParams = formatParams(params, module, funcCaller);
+    winstonLogger.error(...formattedParams);
+}
+
+function warnAndFuncCaller(funcCaller, module, ...params) {
+    const formattedParams = formatParams(params, module, funcCaller);
+    winstonLogger.warn(...formattedParams);
+}
+
+function debugLog(module, ...params) {
+    if (!winstonLogger.isLevelEnabled('debug')) {
+        return;
+    }
+    debugAndFuncCaller(getFunctionData(debugLog), module, ...params);
+}
+
+function infoLog(module, ...params) {
+    if (!winstonLogger.isLevelEnabled('info')) {
+        return;
+    }
+    const formattedParams = formatParams(params, module);
+    winstonLogger.info(...formattedParams);
+}
+
+function errorLog(module, ...params) {
+    if (!winstonLogger.isLevelEnabled('error')) {
+        return;
+    }
+    errorAndFuncCaller(getFunctionData(errorLog), module, ...params);
+}
+
+function warnLog(module, ...params) {
+    if (!winstonLogger.isLevelEnabled('warn')) {
+        return;
+    }
+    warnAndFuncCaller(getFunctionData(warnLog), module, ...params);
+}
+
+const _consoleLog = console.log;
+const _consoleError = console.error;
+const _consoleWarn = console.warn;
+
+const sentryMessageToAvoid = 'Sentry Logger';
+const sentryMessageType = 'string';
+
+if (process.env.NODE_ENV !== 'test') {
```

Comment:
I would like to invert it...
```suggestion
if (process.env.NODE_ENV === 'test') {
    return;
 }
console.log.......    
```